202011.00 Release changelog revision (#2720)
aws/amazon-freertos: CHANGELOG.md

- Added new MQTT and shadow demos for BLE. The demo uses the new MQTT and Shadow client libraries and the BLE transport interface to send and receive data with AWS IoT.

> - Added the SHIM network interface for backwards compatiblity of existing MQTT, Shadow and OTA demos with Bluetooth Low Energy.

Reviewer: ```suggestion
- Provides backwards compatibility for application existing MQTT, Shadow and OTA libraries with Bluetooth Low Energy.
```

- Added FreeRTOS Console API to interact with CLI over common IO or UDP interface.
- Added sample which demonstrates executing commands using FreeRTOS+CLI and UART interface.

#### FreeRTOS+TCP V2.4.0

> - Sub-moduled to the [FreeRTOS/FreeRTOS-Plus-TCP](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP) repository

Reviewer 2: See the modified text below.

- Sub-moduled to the [FreeRTOS/FreeRTOS-Plus-TCP](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP) repository. This is a breaking change for customer using `202007` release of amazon-freertos due to change in folder structure. The version bump was done in the single source of truth of FreeRTOS+TCP (`V2.2.2` -> `V2.3.0`). Since the version number in amazon-freertos is already at `2.3.0` (discrepancy due to multiple sources of truth), it is receiving a minor version bump to `2.3.1`. 

@gitwallit, any comments? I tried to make this descriptive for clarity.

Reviewer 3: single source of truth may not be familiar with other viewers. we could perhaps say - "The version bump for this change was done in the FreeRTOS-Plus-TCP repository (V2.2.2 -> V2.3.0). Since the version number in the amazon-freertos repository is already at 2.3.0 (this discrepancy is caused by the presence of multiple FreeRTOS+TCP sources, which has since been corrected with the FreeRTOS-Plus-TCP repository as the single source of truth), only a minor version bump to 2.3.1 is being done.

Reviewer 2: In essence, the whole thing should read:

"Sub-moduled to the [FreeRTOS/FreeRTOS-Plus-TCP](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP) repository. This is a breaking change for users using `202007` release of amazon-freertos due to change in folder structure. The version bump for this change was done in the FreeRTOS-Plus-TCP repository (V2.2.2 -> V2.3.0). Since the version number in the amazon-freertos repository is already at 2.3.0 (this discrepancy is caused by the presence of multiple FreeRTOS+TCP sources, which has since been corrected with the FreeRTOS-Plus-TCP repository as the single source of truth), only a minor version bump to 2.3.1 is being done."

Is this accurate?

- Refactored as shim layer for V2.x.x MQTT APIs using CoreMQTT library.

#### Over the Air Update V1.2.1

- Added check to abort the update if updating job status as self-test in service fails, this helps in early detection of mismatch in device and jobs states before activating new image.

#### PKCS11 V2.1.1

- Sub-moduled to the [FreeRTOS/corePKCS11](https://github.com/FreeRTOS/corePKCS11) repository.
- Updated ECC608A and PSA library dependencies to corePKCS naming scheme. Refactored AFQP tests to support HSMs that have locked down credentials.

#### Secure Sockets LwIP V1.3.0

- Added new optional API `SOCKETS_Bind`.
- Extended `SOCKETS_SetSockOpt` to support TCP keepalive settings.

#### Shadow V2.2.3

- Updated unit tests to work with the MQTT shim.

#### TLS V1.3.0

- Added missing Max Fragment Length (MFL) runtime configuration if MFL macro is enabled.

#### Wi-Fi V2.0.0

> - Updated WiFi API signatures to make it more granular and to support more features in future.

Reviewer: ```suggestion
- Updated WiFi APIs  to support more granular error codes,  WEP encryption, SoftAP provisioning , Event handling and set country code. 
```
